### PR TITLE
Switch to content scripts

### PR DESCRIPTION
**Pull request by banocean**
Fixes https://github.com/banocean/ifv/issues/46

### CODE REVIEW

File: apply.js [hunk 1]
```diff
@@ -137,8 +16,37 @@ const allowedHostnames = [
...
+
+const getPatchesFiles = (patches, config) => {
+  return patches.flatMap((patch) => {
+    const result = []
+    if (config[patch.name].enable) {
+      if (patch.files?.js?.length) result.push(patch.files.js.map((file) => `patches/${file}`));
+      if ((!patch.allowedHostsCss || patch.allowedHostsCss.includes(window.location.hostname)) && patch.files?.css?.length) {
+        result.push(patch.files.css.map((file) => `patches/${file}`))
+      }
+    }
```
Comment (Marioneq4958) [inline apply.js:41]:
```suggestion
    if (!config[patch.name].enable) return;
    if (patch.files?.js?.length) result.push(patch.files.js.map((file) => `patches/${file}`));
    if ((!patch.allowedHostsCss || patch.allowedHostsCss.includes(window.location.hostname)) && patch.files?.css?.length) {
        result.push(patch.files.css.map((file) => `patches/${file}`))
    }
```

File: execute.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        onlyOnReloads,
+        doesRunHere,
+        isLoaded,
+        run,
+    } of modules) {
+        if (onlyOnReloads && !isFirstRun) continue;
+        if (doesRunHere !== undefined && !doesRunHere()) continue;
+
+        if (isLoaded === undefined) run();
+        else if (isLoaded()) run();
```
Comment (Marioneq4958) [inline execute.js:29]:
```suggestion
        if (isLoaded === undefined || isLoaded()) run();
```

**Review by Marioneq4958 — CHANGES_REQUESTED** [review]
There are a few things to improve. Add missing semicolons.

File: apply.js (hunk 1 above)
Comment (banocean) [reply apply.js:41]: Your code suggestion even introduces an error, there is `return result;` after that for a reason and it's as readable as this

**Review by Marioneq4958 — APPROVED** [review]
All right, good job 🎉

**Review by PanLiszka — APPROVED** [review]
Seems to work totally fine. Good job! 👏

**Pull request merged**